src/ray/raylet/node_manager.cc

```diff
@@ -294,26 +295,30 @@ void NodeManager::ProcessNewClient(LocalClientConnection &client) {
 
 void NodeManager::DispatchTasks() {
   // Work with a copy of scheduled tasks.
-  auto scheduled_tasks = local_queues_.GetScheduledTasks();
+  const auto& scheduled_tasks = local_queues_.GetScheduledTasks();
   // Return if there are no tasks to schedule.
   if (scheduled_tasks.empty()) {
     return;
   }
   const ClientID &my_client_id = gcs_client_->client_table().GetLocalClientId();
-
-  for (const auto &task : scheduled_tasks) {
-    const auto &local_resources =
+  const auto &local_resources =
         cluster_resource_map_[my_client_id].GetAvailableResources();
+  
+  std::unordered_set<TaskID> dispatched_taskIDs;
+  for (const auto &task : scheduled_tasks) {
     const auto &task_resources = task.GetTaskSpecification().GetRequiredResources();
     if (!task_resources.IsSubset(local_resources)) {
       // Not enough local resources for this task right now, skip this task.
       continue;
     }
-    // We have enough resources for this task. Assign task.
-    // TODO(atumanov): perform the task state/queue transition inside AssignTask.
-    auto dispatched_task =
-        local_queues_.RemoveTasks({task.GetTaskSpecification().TaskId()});
-    AssignTask(dispatched_task.front());
+    dispatched_taskIDs.insert(task.GetTaskSpecification().TaskId());
```

Comment:
in terms of correctness, it should be fine to batch dispatched ids and then dispatch them. The intention was to dispatch them ASAP, to minimize latency. I'm ok with this change for now.